### PR TITLE
Implement linked_selection.filter method

### PR DESCRIPTION
**Pull request by philippjfr**
Simple helper for applying a linked_selection to some data.

### CODE REVIEW

File: holoviews/selection.py [hunk 1]
```diff
@@ -325,6 +325,24 @@ def selection_param(self, data):
...
+        Args:
+            data: A Dataset type or data which can be cast to a Dataset
+
+        Returns:
+            The filtered data
+        """
+        if self.selection_expr is None:
+            return data
+        if not isinstance(data, Dataset):
+            raw = True
```
Comment (jbednar) [inline holoviews/selection.py:342]: What does this line achieve?

Comment (philippjfr) [reply holoviews/selection.py:342]: Nothing, good catch, was rushed and copy pasted.

Comment (philippjfr) [reply holoviews/selection.py:342]:
```suggestion
```

File: holoviews/selection.py [hunk 2]
```diff
@@ -325,6 +325,23 @@ def selection_param(self, data):
...
+            data: A Dataset type or data which can be cast to a Dataset
+
+        Returns:
+            The filtered data
+        """
+        if self.selection_expr is None:
+            return data
+        if not isinstance(data, Dataset):
+            data = Dataset(data)
+        return data[self.selection_expr.apply(Dataset(data))]
```
Comment (philippjfr) [inline holoviews/selection.py:343]:
```suggestion
        return data[self.selection_expr.apply(data)]
```

**Pull request merged**